### PR TITLE
Fix #1925: Add Github Actions test workflow

### LINKED ISSUE #1925: Convert CI to use Github Actions
From my tests it looks like using GH actions gives us 2-3 shorter test runs.

### PR DESCRIPTION
**Pull request by iccub**
## Summary of Changes

This pull request fixes issue #1925 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Review by kylehickinson — CHANGES_REQUESTED** [review]
The `build_and_test.yml` needs to be in the `.github` folder :)

**Comment by iccub** [comment]
Weird, I added it via Actions panel

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        uses: actions/cache@v1
+        with:
+          path: Carthage
+          # New XCode/swift version requires recompiling Carthage dependencies.
+          key: carthage-${{ env.XCODE_VERSION }}-${{ hashFiles('**/Cartfile.resolved') }}
+          restore-keys: |
+            carthage-${{ env.XCODE_VERSION }}-
+            carthage-
+      - name: Run bootstrap script
+        run: ./bootstrap.sh -ci
```
Comment (kylehickinson) [inline .github/workflows/build_and_test.yml]: `-ci` should be `--ci`

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+name: Build
+
+on: pull_request
+
+jobs:
+  test:
+    name: test
```
Comment (kylehickinson) [inline .github/workflows/build_and_test.yml]: Maybe `Run Tests` or `Test`? (This one shows up in the PR check name)

**Review by kylehickinson — CHANGES_REQUESTED** [review]

**Review by kylehickinson — APPROVED** [review]

**Pull request merged**